xdsl/interactive/app.py

```diff
@@ -394,45 +406,55 @@ def update_input_operation_count_tuple(self, input_module: ModuleOp) -> None:
         Function that updates the input_operation_datatable to display the operation
         names and counts in the input text area.
         """
+        # sort tuples alphabetically by operation name
         self.input_operation_count_tuple = tuple(
-            count_number_of_operations(input_module).items()
+            (k, v)
+            for (k, v) in sorted(count_number_of_operations(input_module).items())
         )
 
     def watch_input_operation_count_tuple(self) -> None:
         """
         Function called when the reactive variable input_operation_count_tuple changes - updates the
         Input DataTable.
         """
+        # clear datatable and add input_operation_count_tuple to DataTable
         self.input_operation_count_datatable.clear()
-        for k, v in self.input_operation_count_tuple:
-            self.input_operation_count_datatable.add_row(k, v)
-
-        self.update_output_operation_count_tuple()
+        self.input_operation_count_datatable.add_rows(self.input_operation_count_tuple)
+        self.update_operation_count_diff_tuple()
 
-    def update_output_operation_count_tuple(self) -> None:
+    def update_operation_count_diff_tuple(self) -> None:
         """
-        Function that updates the output_operation_datatable to display the operation
-        names and counts in the output text area. It also displays the diff of the input
-        and output datatable.
+        Function that updates the diff_operation_count_tuple to calculate the diff
+        of the input and output operation counts.
         """
         match self.current_module:
             case None:
-                self.output_operation_count_tuple = ()
+                self.diff_operation_count_tuple = get_diff_operation_count(
+                    self.input_operation_count_tuple, ()
+                )
             case Exception():
-                self.output_operation_count_tuple = ()
+                self.diff_operation_count_tuple = get_diff_operation_count(
+                    self.input_operation_count_tuple, ()
+                )
             case ModuleOp():
-                self.output_operation_count_tuple = tuple(
-                    count_number_of_operations(self.current_module).items()
+                # sort tuples alphabetically by operation name
+                output_operation_count_tuple = tuple(
+                    (k, v)
+                    for (k, v) in sorted(
+                        count_number_of_operations(self.current_module).items()
+                    )
+                )
+                self.diff_operation_count_tuple = get_diff_operation_count(
+                    self.input_operation_count_tuple, output_operation_count_tuple
                 )
```

Comment:
I'd recommend setting the output_operation_count_tuple inside the switch, to an empty tuple if necessary, and using that value after the switch.
```suggestion
                # sort tuples alphabetically by operation name
                output_operation_count_tuple = tuple(
                    (k, v)
                    for (k, v) in sorted(
                        count_number_of_operations(self.current_module).items()
                    )
                )
            self.diff_operation_count_tuple = get_diff_operation_count(
                self.input_operation_count_tuple, output_operation_count_tuple
            )
```